### PR TITLE
DP-24265 Modify heading levels for event items in event listings

### PR DESCRIPTION
**Pull request by ygannett**
Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/develop/docs/for-developers/changelog-instructions.md)

## Description

- Adjust the condition to set a heading level for event items in packages/patternlab/styleguide/source/_patterns/03-organisms/by-author/event-listing.twig.
     - Change the heading level of event items in the Upcoming Events in org. page from `h2` to `h3`.
- Change the heading level of event items in the event listing page from `h3` to `h2`.

## Related Issue / Ticket

- [JIRA issue](https://massgov.atlassian.net/browse/DP-24265)
- [Github issue]()

## Steps to Test

1.  Go to /patterns/05-pages-organization-boards/05-pages-organization-boards.html.
2. In the Upcoming Events section, find the event items' headings are `h3` under its comp heading of `h2`.
3. Go to /patterns/05-pages-event-listing/05-pages-event-listing.html?_page=1
4. Find the event items' headings are `h2` following the page title of `h1`.
5. Find no visual change in both pages.

## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

## Additional Notes:

Anything else to add?

#### Impacted Areas in Application

*

#### @TODO

*

### CODE REVIEW

File: packages/patternlab/styleguide/source/assets/js/templates/eventListingRow.html [hunk 1]
```diff
@@ -17,12 +17,12 @@
     {{/if}}
     <div class="ma__event-teaser__content">
       {{#if title.text}}
-      <h3 class="ma__event-teaser__title">
+      <h2 class="ma__event-teaser__title">
```
Comment (clairesunstudio) [inline packages/patternlab/styleguide/source/assets/js/templates/eventListingRow.html]: Shouldn't this be h3? I'm seeing this as h2 
Screen Shot 2022-05-26 at 12 46 22 PM

**Comment by clairesunstudio** [comment]
I'm seeing similar heading level issues all over the place, which makes me think whether it's worth it to use heading for the teasers. I think teasers or more commonly named "cards" don't use a heading for their titles, I think we should just wrap it in `<a>` without nesting them in headings. What do you think?

A few other places I noticed:

This is a h2 under an h3 section heading 
https://mayflower.digital.mass.gov/patternlab/b/patternlab/DP-24265_upcoming-events-heading-level/?p=pages-organization-boards

Screen Shot 2022-05-26 at 12 52 09 PM

And this is on homepage h5 directly under h3
Screen Shot 2022-05-25 at 11 10 59 AM

File: packages/patternlab/styleguide/source/assets/js/templates/eventListingRow.html (hunk 1 above)
Comment (ygannett) [reply packages/patternlab/styleguide/source/assets/js/templates/eventListingRow.html]: You're right.  I missed the result title. The teaser title is back to `h3`.

**Comment by ygannett** [comment]
@clairesunstudio We have 2 things to consider with this.

First, the title of a card can be linked or not linked.  So, we cannot simply replace the heading with `<a>`.

Second, screen readers have a couple features with headings to help their users navigate and/or understand a structure of presented content.  One of them lets users jump to a next heading. As a card has its title heading, screen reader users can navigate between cards through their headings without going through all content in cards.  I checked some accessible design pattern libraries for a card.  They use a heading for its title.

They can can navigate between links, but if content of cards contains links, this feature wouldn't be as helpful as heading one.

For us, it's easier to maintain with `<a>`, but I think it's better to use headings for assistive technology users.  We still need to set up heading levels correctly to better serve them, though.

**Comment by ygannett** [comment]
The backstop failure is not relevant:  https://output.circle-artifacts.com/output/job/4abded70-58b8-4995-8a20-fd20f6dae567/artifacts/0/backstop/reports/html/index.html
The map is slightly shifted inside the iframe from its reference.

**Comment by clairesunstudio** [comment]
ok, I'm fine with not changing the heading tag for now. I have just been noticing the heading level semantic wrong in different places, as I mentioned earlier, and on the homepage. I will create separate tickets for those. 

FYI, I was looking at the semantic UI library and their card title is not a heading https://semantic-ui.com/views/card.html
They were using `<a>` for links and `<div>` for ones not linked.

**Comment by clairesunstudio** [comment]
See followup ticket here https://massgov.atlassian.net/browse/DP-25078

**Review by clairesunstudio — APPROVED** [review]

**Pull request merged**